File: 5_ABSTRACT_DATA/Java/Stack_Parametrico/StackMain.java
import java.io.*;

public class StackMain {

	public static Integer sum(Integer a, Integer b) {
		Integer s;
		s = a + b;
		return s;
	}

	public static void main(String[] args) {
		Stack<Integer> myStack;
		if (args.length <= 0) return;

		myStack = new Stack<>(100);

		for (int i = 0; i < Integer.parseInt(args[0]); i++) 
			myStack.push(i);

		while (!myStack.empty()) {
			System.out.println("Stack top is: " + myStack.top());
			myStack.pop();
		}

		System.out.println(sum(1, 1));

		return;
	}
}

File: 5_ABSTRACT_DATA/C/stack_main.c
#include <stdio.h>
#include <stdlib.h>

#include "stack.h"

int main(int argc, char ** argv) {
Stack s;
int i;

	if (argc <= 1) return -1;

	create(&s);
	
	for ( i = 0; i<atoi(argv[1]); i++) push(&s, i);

	while ( empty(s) == 0 ) {
		printf("Stack top is: %d\n" , top(s) );
		pop(&s);
	}

	destroy(&s);

	return 0;
}
